src/inmanta/agent/handler.py

```diff
@@ -551,28 +551,49 @@ def _should_reload() -> bool:
                 raise Exception(f"Failed to determine whether resource should reload{error_msg_from_server}")
             return result.result["data"]
 
-        def filter_resources_in_unexpected_state(
-            reqs: abc.Mapping[ResourceIdStr, ResourceState]
+        def filter_resources_by_state(
+            reqs: abc.Mapping[ResourceIdStr, ResourceState], states: typing.Set[const.ResourceState]
         ) -> abc.Mapping[ResourceIdStr, ResourceState]:
             """
-            Return a sub-dictionary of reqs with only those resources that are in an unexpected state.
+            Return a sub-dictionary of dependencies of this resource.
+            Only keeping dependencies that are in a state that is in the provided set.
+
+            :param reqs: The list of requirements of this resource
+            :param states: The list of states that we want to keep in this sub-dictionary
             """
-            unexpected_states = {
-                const.ResourceState.available,
+
+            return {rid: state for rid, state in reqs.items() if state in states}
+
+        # Check if any dependencies got into any unexpected state
+        dependencies_in_unexpected_state = filter_resources_by_state(
+            requires,
+            {
                 const.ResourceState.dry,
                 const.ResourceState.undefined,
                 const.ResourceState.skipped_for_undefined,
-                const.ResourceState.deploying,
-            }
-            return {rid: state for rid, state in reqs.items() if state in unexpected_states}
-
-        resources_in_unexpected_state = filter_resources_in_unexpected_state(requires)
-        if resources_in_unexpected_state:
+            },
```

Comment:
Are the other states, by definition, unexpected? 
i.e. unavailable / skipped / failed / cancelled ?

Reply:
These states should be processed by the handler, because they are deploy-time failures states. The lsm transfer resource will inspect the states of its dependencies to determine whether it should move the service into a success or a failed state.